device/iot-device-client/src/main/java/com/microsoft/azure/sdk/iot/device/DeviceIO.java

```diff
@@ -452,28 +455,17 @@ void registerMultiplexingConnectionStateCallback(IotHubConnectionStatusChangeCal
     @Override
     public void execute(IotHubConnectionStatus status, IotHubConnectionStatusChangeReason statusChangeReason, Throwable throwable, Object callbackContext)
     {
-        synchronized (this.stateLock)
```

Comment:
There is no need to synchronize on this lock in this callback since it is either called by the user's thread that called ```open```/```close``` (which already grabbed this lock in this deviceIO layer), or it is called by the reconnection thread. When it is called by the reconnection thread, we know that there isn't a ```close``` call happening since we have the ```reconnectionLock``` in the IotHubTransport layer preventing exactly that.